src/Dialect/ONNX/ONNXOps.cpp

```diff
@@ -867,6 +867,61 @@ LogicalResult ONNXSeluOp::inferShapes(
   return success();
 }
 
+LogicalResult ONNXSequenceInsertOp::inferShapes(
+    std::function<void(mlir::Region &)> doShapeInference) {
+  return success();
+}
+
+static LogicalResult verify(ONNXSequenceInsertOp op) {
+  ONNXSequenceInsertOpAdaptor operandAdaptor = ONNXSequenceInsertOpAdaptor(op);
+
+  // These cast should be guaranteed by default verifier
+  auto seqElementType = operandAdaptor.input_sequence()
+                            .getType()
+                            .dyn_cast<mlir::onnxmlir::SeqType>()
+                            .getElementType();
+  auto elementType1 = seqElementType.dyn_cast<ShapedType>().getElementType();
```

Comment:
Suggest to use  `Type` rather than auto.

Reply:
I agree, using types when its not crazy complicated (e.g. lambda) is best, think its LLVM policy anyway.

Reply:
Fixed